cmd/stack_create.go

```diff
@@ -0,0 +1,182 @@
+			}
+
+			stack := args[0]
+
+			match, err := validateKubernetesResourceName(stack)
+
+			if !match {
+				return err
+			}
+			exists, err := Exists(stack)
+
+			if err != nil {
+				return err
+			}
+
+			if exists {
+				return errors.New("A stack named" + stack + "already exists in your directory. Specify a unique stack name")
+			}
+
+			err = downloadFileToDisk("https://github.com/appsody/stacks/archive/master.zip", filepath.Join(getHome(rootConfig), "extract", "repo.zip"), config.Dryrun)
+			if err != nil {
+				return err
+			}
+			_, projectType, err := parseProjectParm(config.copy, config.RootCommandConfig)
+			if err != nil {
+				return err
+			}
+
+			valid, unzipErr := unzip(filepath.Join(getHome(rootConfig), "extract", "repo.zip"), stack, config.copy)
+
+			if unzipErr != nil {
+				return unzipErr
+			}
+
+			if !valid {
+				return errors.Errorf("Invalid stack name:" + stack + ". Stack name must be in the format <repo>/<stack>")
+			}
+
+			//deleting the stacks repo zip
+			os.Remove(filepath.Join(getHome(rootConfig), "extract", "repo.zip"))
+
+			//moving out the stack which we need
+			err = os.Rename(filepath.Join(stack, "stacks-master", config.copy), projectType)
+			if err != nil {
+				return err
+			}
+
+			//deleting the folder from which stack is extracted
+			os.RemoveAll(stack)
+
+			//rename the stack to the name which user want
+			err = os.Rename(projectType, stack)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	stackCmd.PersistentFlags().StringVar(&config.copy, "copy", "samples/sample-stack", "Copy existing stack")
+	return stackCmd
+}
+
+// Unzip will decompress a zip archive
+// within the zip file (parameter 1) to an output directory (parameter 2).
+func unzip(src string, dest string, copy string) (bool, error) {
+	valid := false
+
+	r, err := zip.OpenReader(src)
+	if err != nil {
+		return valid, err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+
+		// Store filename/path for returning and using later on
+		fpath := filepath.Join(dest, f.Name)
+
+		// Check for ZipSlip. More Info: http://bit.ly/2MsjAWE
+		if !strings.HasPrefix(fpath, filepath.Clean(dest)+string(os.PathSeparator)) {
+			return valid, errors.Errorf("%s: illegal file path", fpath)
+		}
+
+		if runtime.GOOS == "windows" {
+			fileName := strings.Replace(f.Name, "/stacks-master", "", -1)
+			if !strings.HasPrefix(fileName, "stacks-master/"+copy+"/") {
+				continue
+			} else {
+				valid = true
+			}
+		} else {
+			fileName := strings.Replace(f.Name, string(os.PathSeparator)+"stacks-master", "", -1)
+			if !strings.HasPrefix(fileName, filepath.Join("stacks-master", string(os.PathSeparator), copy)+string(os.PathSeparator)) {
+				continue
+			} else {
+				valid = true
+			}
+		}
+
+		if f.FileInfo().IsDir() {
+			// Make Folder
+			err := os.MkdirAll(fpath, os.ModePerm)
+			if err != nil {
+				return valid, err
+			}
+			continue
+		}
+
+		// Make File
```

Comment:
Just a suggestion, we create a separate function in the utils for this: such as`copyFile(src string, des, string)`.